### PR TITLE
fix: Fixes inability to use duplicate names across multiple actions in different pages

### LINKED ISSUE #7301: [Bug]Duplicate query name should be allowed across pages
## Description
All actions (Queries & APIs) should have unique names within a page but can be repeated across different pages in the same application.

Queries and APIs can either be renamed in two ways:
One , from the edit window, where you would click on the current name of the query/API to rename it.
And two, from the entity explorer (the left hand side navigation pane), by clicking on the menu options for that query/API (see video). 

The refactor in the edit window functions correctly for repeated names across pages, but the edit option on the entity explorer errors out for the same value.

The expected behaviour is that the validation for the name refactor from the entity explorer also passes for such scenarios.

### Steps to reproduce the behaviour:

1. Create a new application with page Page1.
2. Create an API in Page1, call it `testAPI`.
3. Create a new page in the same application called Page2.
4. Create another API, this time in Page2.
5. Try to rename this new API to `testAPI` from the entity explorer.
6. See validation error.

[LOOM DEMO](https://www.loom.com/share/4ee46313707046ab944fe420805618b6)

### Important Details

- Version: [Cloud ]
- Browser [chrome]
- Environment [ release]

## Contribution

**Guidelines**
Please read the [Code Contribution Guidelines](https://github.com/appsmithorg/appsmith/blob/master/contributions/CodeContributionsGuidelines.md) to understand the process to get started with solving a Github issue.

**How to fix**
- Currently the logic to find conflict in action names looks for all the action names across all the pages in an application.
- Instead of looking for action names across all the pages, only look for action names within the page. 

**Relevant files and methods**
- [ActionNameEditor.tsx](https://github.com/appsmithorg/appsmith/blob/09eea59330df96c2acbbefa1fcc401ff8026ee12/app/client/src/components/editorComponents/ActionNameEditor.tsx)
  - Please take a look at the `hasActionNameConflict` method for the **correct** implementation. 
- [Name.tsx](https://github.com/appsmithorg/appsmith/blob/b49f5097c6e02ff42ae493416a1a9f141d9ed5eb/app/client/src/pages/Editor/Explorer/Entity/Name.tsx)
  - Please modify the method `hasNameConflict`method. Existing selector (`getExistingActionNames`) used in the file `Name.tsx` is fetching all the action names from all the pages. The selector needs to be changed to only fetch action names from the current page like the `actions` selector in `ActionNameEditor.tsx`.
  - You may refer to this older PR for help: https://github.com/appsmithorg/appsmith/pull/1781 

**Test Cases**
Please write test cases that test the following scenarios:
- Conflicting action name on the same page should give error.
- Same action name on different pages should **not** give any error.
- Please add the test cases in [this](https://github.com/appsmithorg/appsmith/blob/71f4a4b0a979f01dd710c02edeae25c09ace6595/app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/ExplorerTests/Entity_Explorer_Entity_Naming_conflict_spec.js) file.
- You may refer to this older PR for help: https://github.com/appsmithorg/appsmith/pull/1781

### PR DESCRIPTION
**Pull request by Irongade**
…ferent pages

## Description
Fixes inability to use duplicate names across multiple actions in different pages. What this means is prior to this PR, it was impossible to use an action name on one page from another page, so if you were in Page 1, you couldn't use the same action name for an action on Page 2. This is against what is expected, Actions should be able to have similar names as long as they are not on the same page.

## Summary of changes made.
I modified the getExistingActionNames selector in the entities selector to be aware of the action being edited and to check if the current page the user is in corresponds to where the action can be found or not, depending on this, we either provide actions name for actions on the page or not on the page.

Fixes #7301

## Type of change
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
Manual Test

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/duplicate_query_name 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.14 **(-0.01)** | 37.09 **(0)** | 33.97 **(0)** | 55.65 **(0)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**
 :green_circle: | app/client/src/selectors/entitiesSelector.ts | 53.52 **(-0.74)** | 12.5 **(2.03)** | 33.07 **(0.01)** | 50.81 **(-1.1)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

File: app/client/src/selectors/entitiesSelector.ts [hunk 1]
```diff
@@ -494,8 +494,39 @@ export const getExistingWidgetNames = createSelector(
 
 export const getExistingActionNames = createSelector(
   (state: AppState) => state.entities.actions,
-  (actions) =>
-    actions.map((action: { config: { name: string } }) => action.config.name),
+  getCurrentPageId,
+  (state: AppState) => state.ui.explorer.entity.editingEntityName,
```
Comment (AmanAgarwal041) [inline app/client/src/selectors/entitiesSelector.ts:498]: Are we getting entity name from here and using it as an entity id in the callback function ? I think some more description as a comment is required

Comment (Irongade) [reply app/client/src/selectors/entitiesSelector.ts:499]: Comments have been added for better understanding and readability.

**Review by rashmi-rai — APPROVED** [review]

**Review by AmanAgarwal041 — APPROVED** [review]
LGTM

**Pull request merged**